### PR TITLE
new array function

### PR DESCRIPTION
**Pull request by lixingwang**
New array functions. 

1. Array.sum 
2. Array.slice
3. Array.reverse
4. Array.merge

To fix issue #110

### CODE REVIEW

File: function/array/descriptor.json [hunk 1]
```diff
@@ -117,6 +117,83 @@
...
+                    "valueType": "any"
+                }
+            ],
+            "return": {
+                "type": "float64"
+            }
+        },
+        {
+            "name": "slice",
+            "description": "Use to get part of array with start index and end index,this selects a half-open range which includes the first element, but excludes the last one",
```
Comment (fm-tibco) [inline function/array/descriptor.json]: Not sure I understand this description. Isn't it sufficient to say that it returns a slice/subset of the array starting at the specified index and ending at the specified index.  Also, should we make the end index optional?  Or maybe consider using  go/python syntax which is much more flexible like "[1:3]", "[:3]", or "[1:]"

Comment (lixingwang) [reply function/array/descriptor.json]: @fm-tibco Yes, it returns a slice/subset.    I made the end index is optional now.   

Thinking that all arguments may come from mapping.  "[1:3]" may not help, right?

This slice function can achieve those cases "[1:3]", "[:3]", or "[1:]" also

Comment (fm-tibco) [reply function/array/descriptor.json]: Maybe reword the description:  
Extracts a part an array by specifying a half-open range with start index and an end index (the element at the end index is excluded)."

File: function/array/slice.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	return []data.Type{data.TypeAny, data.TypeInt, data.TypeInt}, false
+}
+
+func (sliceFunc) Eval(params ...interface{}) (interface{}, error) {
+	array := params[0]
+	start, err := coerce.ToInt(params[1])
+	if err != nil {
+		return nil, fmt.Errorf("array slice second arguments must be a integer")
+	}
+	len, err := coerce.ToInt(params[2])
```
Comment (fm-tibco) [inline function/array/slice.go]: Is this correct, I thought we were specifying a half-open range, and not a start and a length.  Also, I though you said the last parameter was optional?

Comment (lixingwang) [reply function/array/slice.go]: sorry, it is start and end.  let me correct the var name.
